aiida/cmdline/commands/cmd_import.py

```diff
@@ -252,104 +104,141 @@ def cmd_import(
-        'archive': '',
-        'group': group,
-        'migration': migration,
-        'extras_mode_existing': ExtrasImportCode[extras_mode_existing].value,
-        'extras_mode_new': extras_mode_new,
-        'comment_mode': comment_mode,
-        'non_interactive': non_interactive,
-    }
-
-    # Import local archives
-    for archive in archives_file:
 
-        echo.echo_info(f'importing archive {archive}')
+def _import_archive(archive: str, web_based: bool, import_kwargs: dict, try_migration: bool):
+    """Perform the archive import.
 
-        # Initialization
-        import_opts['archive'] = archive
-        import_opts['file_to_import'] = import_opts['archive']
-        import_opts['more_archives'] = archive != archives_file[-1] or archives_url
+    :param archive: the path or URL to the archive
+    :param web_based: If the archive needs to be downloaded first
+    :param import_kwargs: key-word arguments to parse to the import function
+    :param try_migration: whether to try a migration if the import raises IncompatibleArchiveVersionError
 
-        # First attempt to import archive
-        migrate_archive = _try_import(migration_performed=False, **import_opts)
-
-        # Migrate archive if needed and desired
-        if migrate_archive:
-            with SandboxFolder() as temp_folder:
-                import_opts['file_to_import'] = _migrate_archive(ctx, temp_folder, silent=False, **import_opts)
-                _try_import(migration_performed=True, **import_opts)
+    """
+    from aiida.common.folders import SandboxFolder
+    from aiida.tools.importexport import (
+        detect_archive_type, EXPORT_VERSION, import_data, IncompatibleArchiveVersionError
+    )
+    from aiida.tools.importexport.archive.migrators import get_migrator
 
-    # Import web-archives
-    for archive in archives_url:
+    with SandboxFolder() as temp_folder:
 
-        # Initialization
-        import_opts['archive'] = archive
-        import_opts['more_archives'] = archive != archives_url[-1]
+        archive_path = archive
 
-        echo.echo_info(f'downloading archive {archive}')
+        if web_based:
+            echo.echo_info(f'downloading archive {archive}')
+            try:
+                response = urllib.request.urlopen(archive)
+            except Exception as exception:
+                _echo_exception(f'downloading archive {archive} failed', exception)
+            temp_folder.create_file_from_filelike(response, 'downloaded_archive.zip')
+            archive_path = temp_folder.get_abs_path('downloaded_archive.zip')
+            echo.echo_success('archive downloaded, proceeding with import')
 
         try:
-            response = urllib.request.urlopen(archive)
+            import_data(archive_path, **import_kwargs)
+        except IncompatibleArchiveVersionError as exception:
+            if try_migration:
+                _echo_exception(f'initial import of {archive} failed, trying migration', exception, warn_only=True)
```

Comment:
Maybe you did this to have information on the archive version included in the message?
Perhaps `IncompatibleArchiveVersionError` could have that version set as an attribute (or you can print the message of hte exception anyhow).

I just feel that a "warning" gives the idea that something is wrong.

Reply:
yeh fair now changed.
it was mainly just left from the original code, the subsequent migration stdout gives info on the versions, so I don't think it is necessary to include that in the message

> Perhaps IncompatibleArchiveVersionError could have that version set as an attribute

yeh good idea, although I'm not going to do that in this PR, there's enough changes already lol